cmd/svcat/broker/register_cmd.go

```diff
@@ -18,57 +18,126 @@ package broker
 
 import (
 	"fmt"
+	"os"
+	"strings"
+	"time"
 
 	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/command"
 	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/output"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	servicecatalog "github.com/kubernetes-incubator/service-catalog/pkg/svcat/service-catalog"
 	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // RegisterCmd contains the information needed to register a broker
 type RegisterCmd struct {
-	BrokerName string
-	Context    *command.Context
-	URL        string
+	*command.Namespaced
+	*command.Waitable
+
+	Context *command.Context
+
+	BasicSecret       string
+	BearerSecret      string
+	BrokerName        string
+	CAFile            string
+	ClassRestrictions []string
+	PlanRestrictions  []string
+	SkipTLS           bool
+	RelistBehavior    string
+	RelistDuration    time.Duration
+	URL               string
 }
 
 // NewRegisterCmd builds a "svcat register" command
 func NewRegisterCmd(cxt *command.Context) *cobra.Command {
 	registerCmd := &RegisterCmd{
-		Context: cxt,
+		Context:    cxt,
+		Namespaced: command.NewNamespaced(cxt),
+		Waitable:   command.NewWaitable(),
 	}
 	cmd := &cobra.Command{
-		Use:   "register NAME --url URL",
```

Comment:
why did we change this?  I'm not sure I really like the positional arguments like this. The `cf` CLI does this, but I think kubernetes gets more into -- args vs positional things and I think we should stick with that.

Reply:
we discussed this at the face-to-face. We decided to switch to positional args as mandatory flags goes against unix command line convention.

Reply:
While this isn't the hill I want to die on 😝, I honestly do not remember that being officially decided at the F2F as you are recalling.

Reply:
@jberkhahn I apologize for this being so weird. I trust that we did decide that and I just forgot the details. Let's go with what you have here and I now crown you master and commander of the CLI so that we don't have to rely on my poor memory anymore!

Reply:
💂

Reply:
hahhahah I was thinking of a 👑 , jeremy immediately went to that ⛵️ movie and I think that's a beefeater? best title ever! 🙌

Reply:
I must have been on one of those bathroom breaks - - I too don't recall the position vs named arg discussion/decision.

Reply:
No? We also discussed this in the weekly meeting the week before the f2f and had the same consensus. We can bring it up again at the meeting next week if you prefer.